src/briefcase/commands/new.py

```diff
@@ -427,29 +450,63 @@ def build_app_context(self):
             "bundle": bundle,
             "url": url,
             "license": project_license,
-            "gui_framework": (gui_framework.split())[0],
         }
 
+    def build_gui_context(self, context: dict[str, str]) -> dict[str, str]:
+        """Build context specific to the GUI toolkit."""
+        bootstraps = get_gui_bootstraps()
+        bootstrap_choices = [
+            "Toga",
+            "PySide2       (does not support iOS/Android deployment)",
+            "PySide6       (does not support iOS/Android deployment)",
+            "PursuedPyBear (does not support iOS/Android deployment)",
+            "Pygame        (does not support iOS/Android deployment)",
+        ]
+        builtin_bootstraps = [n.split(" ")[0] for n in bootstrap_choices]
+        bootstrap_choices += [f for f in bootstraps if f not in builtin_bootstraps]
+        bootstrap_choices.append("None")
+
+        bootstrap_choice = self.input_select(
+            intro="""
+What GUI toolkit do you want to use for this project?""",
+            variable="GUI framework",
+            options=bootstrap_choices,
+        )
+
+        gui_context = {}
+
+        if bootstrap_choice != "None":
+            try:
+                bootstrap_class = bootstraps[bootstrap_choice]
+            except KeyError:
+                bootstrap_class = bootstraps[bootstrap_choice.split(" ")[0]]
+
+            bootstrap = bootstrap_class(context=context)
+
+            if hasattr(bootstrap, "additional_context"):
+                if (additional_context := bootstrap.additional_context()) is not None:
+                    gui_context.update(additional_context)
```

Comment:
I know the preference is to catch `AttributeError` instead of using `hasattr()`....but it made the code quite cumbersome.

```python
try:
    additional_context = bootstrap.additional_context
except AttributeError:
    pass
else:
    if (value := additional_context()) is not None:
        gui_context.update(value)
```
I felt it necessary to do all this to avoid squashing an `AttributeError` from the plugin.

So, I just went with `hasattr()`.